File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minn = prices[0];
        vector<int> dp(prices.size(), 0);
        int ans = 0;
        for(int i = 0; i<prices.size(); i++){
            if(prices[i]>minn){
                int temp = prices[i]-minn;
                ans = max(ans,temp);
            }
            minn = min(minn, prices[i]);
        }
        return ans;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minn = prices[0]
        dp = [0]*len(prices)
        ans = 0
        for i in range(len(prices)):
            if prices[i]>minn:
                ans = max(ans, prices[i] - minn)
            minn= min(minn, prices[i])
        return ans